lib/streamlit/elements/multiselect.py

```diff
@@ -104,6 +104,39 @@ def _check_and_convert_to_indices(
     return [opt.index(value) for value in default_values]
 
 
+def get_default_count(default: Union[Sequence[Any], Any, None]) -> int:
+    if default is None:
+        return 0
+    if not is_iterable(default):
+        return 1
+    return len(cast(Sequence[Any], default))
+
+
+def getOptionsMessage(current_selections: int, max_selections: int):
+    middle_msg_part = f"""but `max_selections`
+is set to {max_selections}. This happened because you either gave too many options to `default`
+or you manipulated the widget's state through `st.session_state`. Note that
+the latter can happen before the line indicated in the traceback."""
+    curr_select_msg = ""
+    max_select_msg = ""
+    if current_selections == 1:
+        curr_select_msg = f"""
+Multiselect has {current_selections} option selected """
+    else:
+        curr_select_msg = f"""
+Multiselect has {current_selections} options selected """
+
+    if max_selections == 1:
+        max_select_msg = f"""
+Please select at most {max_selections} option.
+"""
+    else:
+        max_select_msg = f"""
+Please select at most {max_selections} options.
+"""
+    return curr_select_msg + middle_msg_part + max_select_msg
```

Comment:
This is really unreadable given that the only thing that needs to differ between the plural and singular forms are whether or not there's an "s" at the end of the word option. It should also be `get_options_message` over `getOptionsMessage` so that it's idiomatic python.

This could be rewritten as something like

```python
def get_options_message(current_selections: int, max_selection: int):
    curr_selections_noun = "option" if current_selections == 1 else "options"
    max_selections_noun = "option" if max_selections == 1 else "options"
    return f"""
Construct the whole error message here in a single format string
"""
"""
```

Reply:
That's way cleaner. thank you! I should've realized that the only difference is the `option` word. I also refactored `Multiselect.tsx`'s `noResultsMsg` to be cleaner because of this comment: 
```
const option =
      this.props.element.maxSelections !== 1 ? "options" : "option"
    return `You can only select up to ${this.props.element.maxSelections} ${option}. Remove an option first.`
```